core/src/labels/labels.cpp

```diff
@@ -221,60 +208,121 @@ void Labels::update(const View& _view, float _dt,
 
     /// Collect and update labels from visible tiles
 
-    m_needUpdate = updateLabels(_styles, _tiles, _dt, dz, _view);
+    updateLabels(_styles, _tiles, _dt, dz, _view);
+
+
+    /// Mark labels to skip transitions
+
+    if (int(m_lastZoom) != int(_view.getZoom())) {
```

Comment:
Give it a try :) But when doing the conflict resolution before checking possible skip transitions the chance is greater to hit the last case in conflict resolution (labels.cpp line:~275)